src/contexts/authContext.tsx

```diff
@@ -0,0 +1,28 @@
+import { createContext, useContext, useState, type ReactNode } from "react";
+
+export const AUTH_COOKIE_KEY = "userId";
+
+type Auth = {
+  isLoggedIn: boolean;
+  userName?: string;
+  userEmail?: string;
+};
+
+type AuthContextType = {
+  auth: Auth;
+  setAuth: (auth: Auth) => void;
+};
+
+const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+export const AuthProvider = ({ children }: { children: ReactNode }) => {
+  const [auth, setAuth] = useState<Auth>({ isLoggedIn: false });
+
+  return <AuthContext.Provider value={{ auth, setAuth }}>{children}</AuthContext.Provider>;
```

Comment:
리액트 19에서는 이렇게 표현해도 된답니다!

```suggestion
  return <AuthContext value={{ auth, setAuth }}>{children}</AuthContext>;
```

Reply:
그리고 지금은 value에 익명 객체를 넘기고 있는데요,
이럴 때 렌더링과 관련해서 어떤 단점이 있을까요?